### PR TITLE
Added Aim and First Shot Variance.

### PR DESCRIPTION
**Pull request by SmartShots**
Adds a random variance of between 0 and 0.2
to both bot aim and first shot delay.

### CODE REVIEW

File: ext/Server/Bot.lua [hunk 1]
```diff
@@ -32,6 +32,8 @@ function Bot:__init(player)
 	self._wayWaitTimer = 0;
 	self._wayWaitYawTimer = 0;
 	self._obstaceSequenceTimer = 0;
+	FirstShotRandomness = math.random (0, 2) / 10;
```
Comment (Joe91) [inline ext/Server/Bot.lua:35]: Lets use MathUtils:GetRandom(from, to) here
This looks nicer and returns directly in float
```suggestion
	FirstShotRandomness = MathUtils:GetRandom(0, 0.2);
```

File: ext/Server/Bot.lua [hunk 2]
```diff
@@ -32,6 +32,8 @@ function Bot:__init(player)
...
+	Config.botFirstShotDelay = Config.botFirstShotDelay + FirstShotRandomness;
```
Comment (Joe91) [inline ext/Server/Bot.lua:36]: Don't write on the Config.vars, as they get used by the UI!! Use something like that instead:
`self._shotTimer = -MathUtils:GetRandom(Config.botFirstShotDelay, Config.botFirstShotDelay + 0.2);`

File: ext/Shared/WeaponModification.lua [hunk 3]
```diff
@@ -112,16 +112,18 @@ function WeaponModification:_ModifyWeapon(p_SoldierWeaponData, index, aimWorseni
 		return;
 	end
 
-	local botAimWorsening = aimWorseningNormal;
+	botAimRandomness = math.random (0, 2) / 10;
```
Comment (Joe91) [inline ext/Shared/WeaponModification.lua:115]: I think we should wait with the aim-modification until its possible to do this for each weapon individually. I will have a look at it...

**Review by Joe91 — CHANGES_REQUESTED** [review]
We can discuss about a better solution...
I would also suggest do add the Randomness as Config-Value, as already discussed...

**Comment by SmartShots** [comment]
I agree, it's a good idea but would be much better if we can find a way to get it to work per BOT or at least to update the weapon on BOT spawn.

**Pull request closed without merging**